### PR TITLE
Fixed integration tests and added 2.2 compatibility

### PR DESCRIPTION
**Pull request by Max-checkmk**
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

- Integration-Tests just seem to run against 2.1 and 2.2, but all 3 sites are created as 2.0
- contact_group, host_group and service_group modules don't work with 2.2
- Integration test for the downtimes-module fails against 2.1 and 2.2 as services (cpu load, memory, ...) aren't found

## What is the new behavior?

- The prep.yml is looped now completely so every site gets created with the installed version from the task before.
- All _group modules have been fixed to work with 2.2
- Changed service description in the downtimes test to existing ones.

## Other information

Fix for _group-modules is not very pretty, but modules should be migrated to new API with version-check anyway.

### CODE REVIEW

File: tests/integration/targets/rule/tasks/prep.yml [hunk 1]
```diff
@@ -22,11 +22,4 @@
...
   register: site_status
   changed_when: site_status.rc == "0"
-  loop: "{{ checkmk_versions }}"
-
-- name: "Gather Date and Time Facts on localhost."
-  ansible.builtin.setup:
-    gather_subset:
-      - date_time
-  delegate_to: localhost
-  run_once: true  # noqa run-once[task]
```
Comment (robin-checkmk) [inline tests/integration/targets/rule/tasks/prep.yml:32]: This cannot be removed. It is necessary for the test to work properly.

**Review by robin-checkmk — CHANGES_REQUESTED** [review]

File: plugins/modules/contact_group.py [hunk 2]
```diff
@@ -184,13 +184,24 @@ def get_current_contact_groups(module, base_url, headers):
...
+                    for el in tmp
+                ]
+            else:  # 2.0.0 and 2.1.0
+                current_groups = [
+                    {
+                        "name": el.get("href").rsplit("/", 1)[-1],
+                        "title": el.get("title", el.get("name")),
+                    }
+                    for el in tmp
+                ]
```
Comment (lgetwan) [inline plugins/modules/contact_group.py:204]: It would be nicer if you could use the getversion() method that you have introduced in #344

**Review by lgetwan — COMMENTED** [review]
Thanks for fixing this!
I'm not sure if you can solve the comment I added in the code without depending on #344, too much.

File: plugins/modules/contact_group.py (hunk 2 above)
Comment (Max-checkmk) [reply plugins/modules/contact_group.py:204]: Yes, but i would prefer to do this when migrating the whole plugin to the new API.

Comment (lgetwan) [reply plugins/modules/contact_group.py:204]: Sounds reasonable.

**Review by lgetwan — APPROVED** [review]
Approved.

**Pull request merged**